src/functions.php

```diff
@@ -147,13 +155,21 @@ function append($stream, $callback, $read_write = STREAM_FILTER_ALL)
  */
 function prepend($stream, $callback, $read_write = STREAM_FILTER_ALL)
 {
-    $ret = @\stream_filter_prepend($stream, register(), $read_write, $callback);
+    $errstr = '';
+    \set_error_handler(function ($_, $error) use (&$errstr) {
+        // Match errstr from PHP's warning message.
+        // stream_filter_prepend() expects parameter 1 to be resource,...
+        $errstr = $error;
+    });
+
+    $ret = \stream_filter_prepend($stream, register(), $read_write, $callback);
+
+    \restore_error_handler();
 
     // PHP 8 throws above on type errors, older PHP and memory issues can throw here
```

Comment:
This suggest PHP 8 type errors aren't currently taken care of as the `stream_filter_prepend()` would throw, can you look into this?

Reply:
Done 👍